rubicon/objc/types.py

```diff
@@ -18,17 +18,194 @@
 __arm64__ = (_any_arm and __LP64__)
 __arm__ = (_any_arm and not __LP64__)
 
-PyObjectEncoding = b'{PyObject=@}'
 
-
-def encoding_for_ctype(vartype):
-    typecodes = {
-        c_char: b'c', c_int: b'i', c_short: b's', c_long: b'l', c_longlong: b'q',
-        c_ubyte: b'C', c_uint: b'I', c_ushort: b'S', c_ulong: b'L', c_ulonglong: b'Q',
-        c_float: b'f', c_double: b'd', c_bool: b'B', c_char_p: b'*', c_void_p: b'@',
-        py_object: PyObjectEncoding
-    }
-    return typecodes.get(vartype, b'?')
+_type_to_ctype_map = {
+    int: c_int,
+    float: c_float,
+    bool: c_bool,
+    # This is not really correct - Python strings are Unicode - but we need to keep this for compatibility.
```

Comment:
Yep, I was thinking of compatibility with previous versions. But if that's not a concern, I'll happily remove that.

If the API isn't stable yet, I would also be in favor of removing the mappings for `int` and `float`, because these Python types don't really match the C types they are currently translated to. Python `int` is arbitrary-precision, while C `int` is (practically always) 32-bit. Python `float` is actually the same precision as C `double`, but it is mapped to C `float`. Both of these can cause unintended loss of precision. And in an Objective-C context, it's more common to use generic typedefs like `NSInteger`, `NSUInteger`, `CGFloat`, etc. rather than a specific primitive C type.